src/main/java/org/dormitory/autobotsoccub/command/txtcommand/ShowPoolCommand.java

```diff
@@ -0,0 +1,39 @@
+package org.dormitory.autobotsoccub.command.txtcommand;
+
+import lombok.AllArgsConstructor;
+import org.dormitory.autobotsoccub.command.TextCommand;
+import org.dormitory.autobotsoccub.command.keyboard.KeyboardFactory;
+import org.dormitory.autobotsoccub.command.result.CommandResult;
+import org.dormitory.autobotsoccub.user.UserPool;
+import org.telegram.telegrambots.api.objects.Update;
+
+import java.util.regex.Pattern;
+
+import static org.dormitory.autobotsoccub.command.keyboard.Button.REGISTER;
+
+@AllArgsConstructor
+public class ShowPoolCommand implements TextCommand {
+    private static final Pattern CMD_PATTERN = Pattern.compile("/showpool");
+
+    private KeyboardFactory keyboardFactory;
+
+    private UserPool userPool;
+
+    @Override
+    public Pattern getCommandPattern() {
+        return CMD_PATTERN;
+    }
+
+    @Override
+    public CommandResult execute(Update update) {
+        if(userPool.isEmptyPool()) {
+            return CommandResult.builder()
+                    .replyMessage("No registered users. You can be first!")
+                    .keyboardMarkup(keyboardFactory.keyboardOf(REGISTER))
+                    .build();
+        }
+        return CommandResult.builder()
+                .replyMessage(userPool.toString())
```

Comment:
[future_notes] UserPool should return _immutable_ pool data that could be converted to command reply message somewhere else.

Reply:
Function is implemented in the UserUtils class